fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/BlockEntitySupportedBlocksRegistry.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.registry;
+
+import java.util.Objects;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.entity.BlockEntityType;
+
+import net.fabricmc.fabric.impl.content.registry.BlockEntityTypeHooks;
+
+/**
+ * A registry that allows mods to modify this list of supported blocks per block entity type. For example, a custom sign
+ * block will not work because Minecraft uses a hardcoded list of blocks that are allowed to host a sign block entity.
+ * Using this registry, custom sign blocks can be registered as valid blocks for a sign block entity.
+ */
+public final class BlockEntitySupportedBlocksRegistry {
+	private BlockEntitySupportedBlocksRegistry() {
+	}
+
+	/**
+	 * Adds additional blocks that the given block entity type supports.
+	 *
+	 * @param type            The block entity type to add support for.
+	 * @param supportedBlocks The blocks to add as additional supported blocks.
+	 */
+	public static void register(BlockEntityType<?> type, Block... supportedBlocks) {
```

Comment:
Could this not just be an injected interface?

Reply:
I thought about it but I felt it was a bit off with the other content registries. It's also because most injected interfaces are there to assist with modded objects/classes (because Mojang loves to make things unnecessarily tricky for mods sometimes), whereas this API alters the behaviour of a vanilla object.

Though, that's how I see it. There were a variety of ways to achieve the same result so I eventually just picked the one that I would be the most comfortable with using.